test-result-summary-client/src/TabularView/TabularView.jsx

```diff
@@ -518,61 +518,57 @@ export default class TabularView extends Component {
         }
     }
 
-    populateCompTable(parserProps) {
+    populateCompTable = async() => {
     	// Each entry is a combination of the data in testData and baselineData, same format two fields benchmarkName and platforms
     	// Now platform entries contain information from both the testJdk and the baselineJdk
         const newArray = [];
-        var metricProps;
-        this.state.testData.forEach(function (testDataObject) {
+        for(let testDataObject of this.state.testData) {
             let consolidatedDataObject = {};
             let benchmark = testDataObject.benchmarkNVM.split(",")[0];
             let metric = testDataObject.benchmarkNVM.split(",")[2];
             consolidatedDataObject.platformsSpecificData = {};
             consolidatedDataObject.benchmarkNVM = testDataObject.benchmarkNVM;
-
-            let higherBetter;
-            
-            //get BenchmarkRouter & Metric files from server
-            metricProps = getMetricProps(parserProps, benchmark, metric);
-            try {
-                if (metricProps["higherbetter"]) {
-                    higherBetter = true;
-                } else {
-                    higherBetter = false;
+            let metricProps, higherBetter
+            //To get the values of highterbetter/units,  first check if Metric does already exist in state, if not get its info from server
+            if(this.state.benchmark) {
+                metricProps = this.state.benchmark["metrics"][metric];
+                higherBetter = metricProps["higherbetter"] ? metricProps["higherbetter"] : true;
+            } else {
+                const propers = await getBenchmarkMetricProps(benchmark);
+                if(propers){
+                    this.setState({benchmark: propers})
                 }
-            } catch (higherBetterNotFoundError) {
-                higherBetter = true;
+                metricProps = "";
```

Comment:
This logic does not work
- the current code only stores the first benchmark. This does not work because each benchmark will have its own props. And the props are different
- If the `this.state.benchmark` cannot be found, then call getBenchmarkMetricProps(). But value such as `higherBetter` does not get set in this case.

On the second thought, we should not use `state` as it is asynchronous. Please set the value in constructor() and store the benchmark key and getBenchmarkMetricProps as key value pair.